lib/translator/inquiry.cc

```diff
@@ -151,6 +151,61 @@ void TranslateUnitSerialNumberVpd(
   WriteValue(result, buffer);
   WriteValue(product_serial_number, buffer.subspan(sizeof(result)));
 }
+
+void TranslateBlockLimitsVpd(const nvme::IdentifyControllerData& identify_ctrl,
+                             absl::Span<uint8_t> buffer) {
+  // The value is in units of the minimum memory
+  // page size (CAP.MPSMIN) and is reported as a power of two (2^n).
+  // A value of 0h indicates that there is no maximum data transfer size
+  uint32_t max_transfer_length;
+  if (identify_ctrl.mdts > 32) {
+    DebugLog("max transfer length is > 2^32");
+    max_transfer_length = 1 << 31;
```

Comment:
There's no need to preserve the power-of-two nature here and you could return the max value for a uint32 (2^32 - 1)

Reply:
Max transfer length should be 2^16 according to section 3.13 of https://www.nvmexpress.org/wp-content/uploads/NVM-Express-SCSI-Translation-Reference-1_1-Gold.pdf